src/message/messagesActions.js

```diff
@@ -31,7 +31,7 @@ export const doNarrow = (newNarrow: Narrow, anchor: number = Number.MAX_SAFE_INT
   const caughtUp = getState().caughtUp[newNarrow] || NULL_CAUGHTUP;
 
   if (!anyMessagesInNewNarrow && !caughtUp.newer && !caughtUp.older) {
-    dispatch(fetchMessagesAtFirstUnread(newNarrow));
+    dispatch(fetchMessagesAtFirstUnread(newNarrow, anchor));
```

Comment:
You can, of course, create a function `fetchMessagesAroundAnchor` that will do the  `config.messagesPerRequest / 2` and other parameters.